python_src/src/lamp_py/performance_manager/l1_rt_trips.py

```diff
@@ -15,46 +15,54 @@
 )
 
 
-# def update_prev_next_trip_stop(db_manager: DatabaseManager) -> None:
-#     """
-#     Update `previous_trip_stop_pk_id` and `next_trip_stop_pk_id` fields in `vehicle_events` table
-#     """
-#     prev_next_trip_stop_cte = (
-#         sa.select(
-#             VehicleEvents.pk_id,
-#             sa.func.lead(VehicleEvents.pk_id)
-#             .over(
-#                 partition_by=VehicleEvents.trip_hash,
-#                 order_by=VehicleEvents.stop_sequence,
-#             )
-#             .label("next_trip_stop_pk_id"),
-#             sa.func.lag(VehicleEvents.pk_id)
-#             .over(
-#                 partition_by=VehicleEvents.trip_hash,
-#                 order_by=VehicleEvents.stop_sequence,
-#             )
-#             .label("previous_trip_stop_pk_id"),
-#         ).join(
-#             TempHashCompare,
-#             TempHashCompare.hash == VehicleEvents.trip_hash,
-#         )
-#     ).cte(name="prev_next_trip_stops")
-
-#     update_query = (
-#         sa.update(VehicleEvents.__table__)
-#         .where(
-#             VehicleEvents.pk_id == prev_next_trip_stop_cte.c.pk_id,
-#         )
-#         .values(
-#             next_trip_stop_pk_id=prev_next_trip_stop_cte.c.next_trip_stop_pk_id,
-#             previous_trip_stop_pk_id=prev_next_trip_stop_cte.c.previous_trip_stop_pk_id,
-#         )
-#     )
-
-#     process_logger = ProcessLogger("l1_trips.update_prev_next_trip_stop")
-#     process_logger.log_start()
-#     db_manager.execute(update_query)
-#     process_logger.log_complete()
+def update_prev_next_trip_stop(db_manager: DatabaseManager) -> None:
+    """
+    Update `previous_trip_stop_pk_id` and `next_trip_stop_pk_id` fields in `vehicle_events` table
+    """
+    temp_trips = (
+        sa.select(
+            TempEventCompare.pm_trip_id,
+        )
+        .distinct()
+        .subquery()
+    )
```

Comment:
we use this so that we're only running the `prev_next_trip_stop_sub` subquery on trips that have been updated with this round of gtfs_rt data?

Reply:
Yes, limits to trips in the `TempEventCompare` table.